dao/project.go

```diff
@@ -228,12 +192,61 @@ func GetUserRelevantProjects(userID int) ([]models.Project, error) {
 	return projects, nil
 }
 
+// GetUserRelevantProjects returns the projects based on publicity and user, disregarding the names etc.
+func GetUserRelevantProjects(userID int, projectName string) ([]models.Project, error) {
+	o := orm.NewOrm()
+
+	sql := `select distinct
+		p.project_id, p.owner_id, p.name,p.creation_time, p.update_time, p.public, pm.role role 
+	 from project p 
+		left join project_member pm on p.project_id = pm.project_id
+	 where p.deleted = 0 and pm.user_id= ?`
+
+	queryParam := make([]interface{}, 1)
+	queryParam = append(queryParam, userID)
+	if projectName != "" {
+		sql += " and p.name like ? "
+		queryParam = append(queryParam, projectName)
+	}
+
+	sql += " order by p.name "
+
+	var r []models.Project
+	_, err := o.Raw(sql, queryParam).QueryRows(&r)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+//GetPublicProjects return all public projects whose name like projectName
+func GetPublicProjects(projectName string) ([]models.Project, error) {
+	o := orm.NewOrm()
+	var publicProjects []models.Project
+	sql := `select project_id, name, public 
```

Comment:
@ywk253100 I have not noticed that, it seems that's enough, I will check and add some fields if necessary.